src/components/autocomplete/js/autocompleteController.js

```diff
@@ -203,7 +207,8 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
       $mdUtil.enableScrolling();
     }
 
-    angular.element($window).off('resize', positionDropdown);
+    angular.element($window).off('resize', onWindowResize);
```

Comment:
What I mean is doing this:

```js
debouncedOnResize = $mdUtil.debounce(function() {
  if (!ctrl.hidden) {
   positionDropdown();
  }
});
```

This still lets you de-register it properly. As for being cleaner, it's probably down to preference.